backend/corpora/lambdas/api/v1/curation/collections/actions.py

```diff
@@ -1,26 +1,41 @@
 from flask import jsonify, g
-
-from .common import reshape_for_curation_api_and_is_allowed, list_collections_curation
+from .common import reshape_for_curation_api_and_is_allowed, add_collection_level_processing_status
 from .common import EntityColumns
-from ......common.corpora_orm import CollectionVisibility
+from ...authorization import is_super_curator
+from ......common.corpora_orm import CollectionVisibility, DbCollection
 from ......common.utils.http_exceptions import UnauthorizedError
 from backend.corpora.api_server.db import dbconnect
 
 
 @dbconnect
-def get(visibility: str, token_info: dict):
+def get(visibility: str, token_info: dict, curator: str = None):
     """
     Collections index endpoint for Curation API. Only return Collection data for which the curator is authorized.
     :param visibility: the CollectionVisibility in string form
     :param token_info: access token info
     @return: Response
     """
-    if not token_info and visibility == CollectionVisibility.PRIVATE.name:
-        raise UnauthorizedError()
-    collections = list_collections_curation(g.db_session, EntityColumns.columns_for_collections, visibility)
-    allowed_collections = []
-    for collection in collections:
-        if reshape_for_curation_api_and_is_allowed(g.db_session, collection, token_info, preview=True):
-            allowed_collections.append(collection)
+    if visibility == CollectionVisibility.PRIVATE.name:
+        if any((not token_info, not is_super_curator(token_info) and curator not in ("self", None))):
+            raise UnauthorizedError()
+
+    filters = [DbCollection.tombstone == False]  # noqa
+    if visibility == CollectionVisibility.PUBLIC.name:
+        filters.append(DbCollection.visibility == CollectionVisibility.PUBLIC)
+    elif visibility == CollectionVisibility.PRIVATE.name:
+        filters.append(DbCollection.visibility == CollectionVisibility.PRIVATE)
+
+    if token_info and curator == "self":  # curator wants own collections
+        filters.append(DbCollection.owner == token_info["sub"])
+    elif curator:  # user want collections from a specific curator
+        filters.append(DbCollection.curator_name == curator)
```

Comment:
👍

Reply:
I think this needs a check for super curator, right? Otherwise what gates a curator from querying for another curator's private collections?

Reply:
a check if visibility is also private, that is

Reply:
nvm, ignore comments above, the check at the top takes care of this